server/src/main/java/io/crate/replication/logical/action/PublicationsStateAction.java

```diff
@@ -128,7 +137,11 @@ protected void masterOperation(Request request,
                     return;
                 }
 
-                List<RelationName> relationNamesOfPublication = resolveRelationsNames(publication, schemas);
+                // Case 'allTables = true' requires additional check for user permissions but looked up user can be null
+                // if user got deleted before any subscription for the publication had been started.
+                // TODO: Either drop publication on owner-user removal or add to the warning a message that publication should be deleted.
```

Comment:
I currently check `user!=null` for both `allTables = true` and `allTables = false`. But I'm not sure about the latter - given that pre-defined case is validated  on publication creation, can we keep replication if publication owner was deleted?

Reply:
Could you check how PG behaves when the publication owner is dropped?

Reply:
In PG it's not possible to remove a user owning a publication (and any object in general)
```
ERROR:  role "newuser" cannot be dropped because some objects depend on it
DETAIL:  owner of publication pub1
```
Basically it correlates with [DROP ROLE docs ](https://www.postgresql.org/docs/current/sql-droprole.html)
> Before dropping the role, you must drop all the objects it owns (or reassign their ownership) 

I'm not sure if we have general check for owned objects on a user removal - is it fine if I add `user!=null` check for both cases - kinda align with PG just on a later stage and throw an Exception to avoid retry?

If yes, will do the same check for subscriber/DQL - if user doesn't exist then throw an Exception.

Reply:
Until now we didn't have an `owner` concept. I think we should do this right and also prevent dropping users to which some objects belongs to.

Reply:
I pushed first part (null checks) + modified request to get subscribing user + updated tests.

>prevent dropping users 
Is it fine to only check pubs/subs (from ClusterMetadata) belonging to the user to reduce the scope? 
There can be views, tables (I probably missed something else) - I will create a new issue for that.

If I add ownership check on drop user, null check in `PublicationsStateAction` probably not needed anymore?

Reply:
I fixed integration tests - checking objects belonging to user is not yet done - just want to verify that in general  approach is on the the right track (especially updating request with the subscriber)

Reply:
> There can be views, tables (I probably missed something else) - I will create a new issue for that.

AFAIK publications are the only ones holding an `owner` entity. Tables, Views etc. do not follower the owner concept.